aikau/src/main/resources/alfresco/documentlibrary/views/AlfGalleryView.js

```diff
@@ -97,11 +96,11 @@ define(["dojo/_base/declare",
        * @param {object}
        */
       updateColumns: function alfresco_documentlibrary_views_AlfGalleryView__updateColumns(payload) {
-         if (this.docListRenderer != null && 
-             payload != null && 
-             payload.value != null &&
+         if (this.docListRenderer && 
+             payload && 
+             payload.value &&
              !isNaN(payload.value) &&
-             this.columns != payload.value)
+             this.columns !== payload.value)
          {
```

Comment:
Lots of repetition here. Might be better with a separate variable first:

```
var newNumCols = payload && !isNaN(payload.value) && payload.value;
if (this.docListRenderer && newNumCols && newNumCols !== this.columns) {
  [log]
  [update]
}
```

Also, this will fail if the new number of columns is zero. That seems reasonable, but can you confirm?